### ISSUE TITLE
accessibility problems and suggestions related to NVDA bot status strings

### ISSUE THREAD

**Issue by mcking65** [w3c/aria-at-app#872]
When there is an active bot run, the status of tests is shown in the test queue. That is helpful. When I open the manage bot run dialog, though, there is no status shown in the dialog. It would be very helpful to show the same information in the dialog.

Like some other places where there is code to determine whether a noun should be singular or plural, the rendering for screen readers of the status strings has problems. Here is how JAWS presents the strings:

>0  Test s  Completed 
>15  Test s  Queued 
>0  Test s  Cancelled 

Note:
* extra space before the word test
* space between 'test' and 's'
* Extra space after 's'

**Comment by jugglinmike** [comment]
Hi @mcking65! I've tackled the bug fix and the feature request in separate patches: https://github.com/w3c/aria-at-app/pull/899 and https://github.com/w3c/aria-at-app/pull/900

**Comment by jugglinmike** [comment]
@mcking65 I'm not able to reproduce the bug using JAWS (version 2024.2312.53) on Windows 10 (Home edition, version 19045.3930) and Firefox (version 122.0) or Edget (version 121.0.2277.83). Can you share your software versions and the steps you are taking to observe the bug?

For my part, when I instruct JAWS to read the line (by pressing the "down" arrow), it says (for instance) "14 tests completed" with no extraneous pauses. When I instruct JAWS to read each character in sequence (by repeatedly pressing the "right" arrow), it says each character with no extraneous spaces.

I do have a theory about what's going wrong, though, and [I've proposed a fix based on that theory](https://github.com/w3c/aria-at-app/pull/900). We'll need to reproduce the problem in order to verify my theory and merge that patch. To that end, would you mind checking if [this reduced test case](http://s92097608.onlinehome.us/tmp/text-nodes.html) exhibits the same problem for you?

**Comment by ccanash** [comment]
Re opening as we referenced this tasks in #899  but it hasn't been addressed

**Comment by mcking65** [comment]
@jugglinmike 

Thank you for the reduced test case and for mentioning that you were testing in Firefox.

This appears to be a bug in JAWS related to Chrome. JAWS is erroneously inserting spaces between the text nodes.

With [that test case](http://s92097608.onlinehome.us/tmp/text-nodes.html):
1. NVDA and Narrator with both Chrome and Firefox read both list items as "Cats". Oddly, in Chrome, Narrator anounces the list items as H1 elements.
2. JAWS+Firefox reads both list items as "Cats".
3. Using Chrome 121 and JAWS version 2024.2310.70.400:
    1. First list item is read as "Cats".
    2. Second list item is read as "Cat s".

Pls package the test case into a zip and attach it, and I will then attach it to a JAWS bug in the JAWS beta program.

If it is not a big deal to workaround this bug by appending whole words, pls do so.

**Comment by jugglinmike** [comment]
> Pls package the test case into a zip and attach it, and I will then attach it to a JAWS bug in the JAWS beta program.

Here you go! [text-nodes.zip](https://github.com/w3c/aria-at-app/files/14364696/text-nodes.zip)

> If it is not a big deal to workaround this bug by appending whole words, pls do so.

Your verification proves that gh-900 fixes this, so we can merge that, now. Please note, however, that the front-end development ecosystem takes the expected behavior for granted. Put differently, tooling and developers all assume that text node boundaries are insignificant. As a result, it will be difficult for us to workaround the JAWS bug consistently without substantial investment in custom tooling for ARIA-AT App. That's why I'm glad to hear you'll be reporting this to JAWS! (Otherwise, if JAWS couldn't or wouldn't fix this, I was considering how to broach the issue to the maintainers of the React framework--the behavior really is that systemic.)
